CIP-XXXX/README.md

```diff
@@ -0,0 +1,114 @@
+
+This CIP proposes an extension to [CIP-13] to allow easy and unambiguous encoding of
+[CIP-0019]/[CIP-0105] Addresses into URL's.
+
+## Motivation: why is this CIP necessary?
+
+[CIP-0013] defines encoding for payment addresses and stake pool, however [CIP-0019]
+and [CIP-0105] define numerous other address types.
+
+These addresses cannot currently be encoded into URIs unambiguously.
+This extension proposes a simple and forward compatible method of encoding such addresses into the URI scheme defined by [CIP-0013].
+
+[x509] certificates can contain name or alternative name information related to either the issuer of
+the certificate or its subject.
+It is desirable to distinguish an Issuer or Subject of a certificate by one or more on-chain keys.
+This, for example, can facilitate the ability to link off-chain actions authorized with a x509 certificate,
+with an on-chain identity.
+
+Currently, there is no standard way to embed a Cardano address, such as a stake address,
+or a dRep address as distinguishing name within a [x509] certificate.
+
+A URI is one of the possible names that can be associated with the Issuer or Subject of a certificate.
+This URI can be referenced in the alternative name for both the Issuer and Subject of a certificate
+by using the uniformResourceIdentifier in the general name.
+[CIP-0013] does not define a method for stake or dRep addresses, etc., to be encoded in the URI scheme it defines.
+
+Allowing these addresses to be easily encoded as URIs allows them to be 100% interoperable
+with existing public key infrastructure and certificate creation tools.
+
+## Specification
+
+We extend [CIP-0013] with a single new authority for referencing Cardano addresses in [CIP-0019] format.
+
+### Grammar & interpretation
+
+We extend the grammar from [CIP-0013] with the new authority:
+
+```
+authorityref = (... | addr )
+
+addr = "//addr/" cip19-addr
+cip19-addr = *cip19-char
+cip19-char = ALPHA / DIGIT / "_"
+```
+
+Effectively, any address string specified by [CIP-0019], [CIP-0105] or future extension to either
+of these specifications can be embedded directly within the URI.
+
+### Examples
+
+```
+web+cardano://addr/addr1qx2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzer3n0d3vllmyqwsx5wktcd8cc3sq835lu7drv2xwl2wywfgse35a3x
+web+cardano://addr/addr1z8phkx6acpnf78fuvxn0mkew3l0fd058hzquvz7w36x4gten0d3vllmyqwsx5wktcd8cc3sq835lu7drv2xwl2wywfgs9yc0hh
+web+cardano://addr/addr1yx2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzerkr0vd4msrxnuwnccdxlhdjar77j6lg0wypcc9uar5d2shs2z78ve
+web+cardano://addr/addr1x8phkx6acpnf78fuvxn0mkew3l0fd058hzquvz7w36x4gt7r0vd4msrxnuwnccdxlhdjar77j6lg0wypcc9uar5d2shskhj42g
+web+cardano://addr/addr1gx2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzer5pnz75xxcrzqf96k
+web+cardano://addr/addr128phkx6acpnf78fuvxn0mkew3l0fd058hzquvz7w36x4gtupnz75xxcrtw79hu
+web+cardano://addr/addr1vx2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzers66hrl8
+web+cardano://addr/addr1w8phkx6acpnf78fuvxn0mkew3l0fd058hzquvz7w36x4gtcyjy7wx
```

Comment:
If this many examples for `//addr/addr1` are included, can you section them to explain what address parameters they each represent?  Most readers can distinguish the stake-key vs. the no-stake-key addresses but might need help with the others.

If some pairs are the same except for a user context (e.g. the same parameters used in 2 different places) it would be more readable only to include the distinct examples.

Reply:
I can certainly trim it.  I just took the examples from CIP-19 which is probably unnecessary, in retrospect, I think all it needs is one example of the main address types from CIP-19 and a selection from CIP-109 to capture the intent.

Reply:
See revised examples section.